src/main/java/map/FileNavigator.java

```diff
@@ -0,0 +1,51 @@
+package map;
+
+import java.util.*;
+
+public class FileNavigator {
+    private final Map<String, List<FileData>> fileMap;
+
+    public FileNavigator() {
+        fileMap = new HashMap<>();
+    }
+
+    public void add(FileData fileData) {
+        String path = fileData.getPath();
+        List<FileData> files = fileMap.getOrDefault(path, new ArrayList<>());
+
+        for (FileData existingFile : files) {
+            if (!existingFile.getPath().equals(path)) {
```

Comment:
Potentially you may have NPE in this way if `existingFile.getPath()==null`

Reply:
updated condition